### PR TITLE
Custom pagination

### PR DESCRIPTION
**Pull request by HectorZR**
# Custom pagination

Reason:
A paginator with the structure:
```json
{
  "ids": [],
  "data": {
    "id": {}
  }
}
```
was needed to optimize frontend processes.

Before pull request:
Pagination was Laravel default.

After pull request:
Now pagination has this structure:
```json
{
  "ids": [],
  "data": {
    "id": {}
  }
}
```

Code Coverage & Build Info
Laravel

### CODE REVIEW

**Review by angel-afonso — CHANGES_REQUESTED** [review]
I love the new implementation, could you add assertions in the test cases where the new paging structure is received? With that change I immediately approve the PR

File: app/Models/Model.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+class Model extends EloquentModel
+{
+  /**
+   * Create a new Eloquent query builder for the model.
+   *
+   * @param  \Illuminate\Database\Query\Builder  $query
+   * @return \Illuminate\Database\Eloquent\Builder|static
+   */
+  public function newEloquentBuilder($query)
+  {
```
Comment (angel-afonso) [inline app/Models/Model.php:17]: I think a command should be created to create models that extend from this base model

Comment (HectorZR) [reply app/Models/Model.php:17]: I modified the base file laravel uses to create new model files. So Laravel is taking this custom model for creating new ones

**Review by angel-afonso — APPROVED** [review]
Very nice :rocket:

**Pull request merged**